### PR TITLE
Switch to using Gradle and the Intellij SDK

### PR DESCRIPTION
**Pull request by maartenba**
Use https://github.com/JetBrains/gradle-intellij-plugin to build the plugin. Make updating to newer Rider versions easier, too.

### CODE REVIEW

**Comment by maartenba** [comment]
Note that the SDK automatically sets plugin since/until versions (and by default, is locked to one major version, in this PR to 2020.2 and updates, but not 2020.3)

File: src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt [hunk 1]
```diff
@@ -7,6 +7,7 @@ import com.intellij.openapi.fileEditor.FileEditorManager
 import com.intellij.openapi.project.DefaultProjectFactory
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.LocalFileSystem
+import com.jediterm.terminal.ui.UIUtil
```
Comment (Witik) [inline src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt]: Why import a class from JediTerm?
The `isWindows` method in that project isn't that different from mine, and it seems unnecessary to add a dependency on an unrelated project just for that method.

I suggest copying the implementation instead, if there's no similar method in the `com.intellij` namespace

**Review by Witik — CHANGES_REQUESTED** [review]
Thank you for this PR! I tried porting it a while back, but never got it working properly.

I left one small comment regarding this PR, and would love to hear your thoughts.

File: src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt (hunk 1 above)
Comment (maartenba) [reply src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt]: Seems like an Alt+Enter gone wrong. Pushed a fix for this.

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+dependencies {
+    implementation "org.jetbrains.kotlin:kotlin-stdlib"
+}
+
+// See https://github.com/JetBrains/gradle-intellij-plugin/
+intellij {
+    type 'RD'
+    version '2020.2'
```
Comment (maartenba) [inline build.gradle:20]: Either a stable version or an EAP snapshot (e.g. `2020.3-EAP3-SNAPSHOT`

Will be used as min version for the plugin, more docs at https://github.com/JetBrains/gradle-intellij-plugin/

**Review by Witik — APPROVED** [review]

**Pull request merged**